src/router_core/connections.c

```diff
@@ -34,6 +34,9 @@ static void qdr_link_delete_CT(qdr_core_t *core, qdr_action_t *action, bool disc
 ALLOC_DEFINE(qdr_connection_t);
 ALLOC_DEFINE(qdr_connection_work_t);
 
+const int AUTO_LINK_FIRST_RETRY_INTERVAL = 2;
```

Comment:
All of this auto-link-retry logic should be in the route_control module, not the connections module.  One hint that this is the case is the fact that you moved qdr_route_log_CT from static to non-static.  qdr_route_log_CT is a static helper function for route_control logic.